zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineApiValidationTask.kt

```diff
@@ -32,13 +32,15 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.internal.file.FileCollectionFactory
 import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFiles
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 
 @Suppress("unused") // Public API for Gradle plugin users.
-abstract class ZiplineApiDumpTask @Inject constructor(
+abstract class ZiplineApiValidationTask @Inject constructor(
   fileCollectionFactory: FileCollectionFactory,
+  @Input val mode: Mode,
 ) : DefaultTask() {
 
   @get:OutputFile
```

Comment:
Does this task behavior properly with up-to-date checking given the file is declared as an output rather than an input?